### PR TITLE
Run integration tests and release jobs on each pull request/branch push

### PR DESCRIPTION
**Pull request by rushiagr**
Signed-off-by: Rushi Agrawal <rushi.agr@gmail.com>

### Description
Noticed integ and release jobs don't work on pull request, so fixed it now
 
### Issues Resolved
- 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -1,16 +1,6 @@
 name: Integration tests
 
-on:
-  pull_request:
-    paths-ignore:
-      - 'README.md'
```
Comment (axeoman) [inline .github/workflows/integration.yml:6]: Ignoring README.md feels like a logical move, but I just wondering why you decided to drop it?

Comment (rushiagr) [reply .github/workflows/integration.yml:6]: Just wanted to keep it simple and consistent with other `.yml` files. The savings (to Github) would be only a small amount, as the PR needs to be exclusively updating README.md to have the CI job not run.

Comment (axeoman) [reply .github/workflows/integration.yml:6]: Got it, thanks!

**Review by axeoman — APPROVED** [review]
Looks good.

**Pull request merged**